gammapy/makers/utils.py

```diff
@@ -465,6 +469,11 @@ def make_theta_squared_table(
 
     create_off = position_off is None
     for observation in observations:
+        if energy_edges is not None:
+            observation = observation.copy()
```

Comment:
Why is the copy necessary?
you can not have 
`event_position = observation.events.select_energy(energy_range=energy_edges).radec
`?